### PR TITLE
Archi rework

### PR DESCRIPTION
**Pull request by DanYellow**
- Splits webpack config. So now there is a base config and prod.config and dev.config inherit from it. It's easier to read config file this way, according to me.
- Adds module resolver to avoid bad relative path with a lot of '../'. Now everywhere in your js file you can start filepath from root.
- Refactor some module resolver

Note : Why do you have postcss-loader AND less-loader. post-css is way more future proof than less.
I'll try to add hot-reload but it looks like nodemode do not like it.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,8 @@
 {
-    "presets": ['env', 'react']
+    "presets": ['env', 'react'],
+    "plugins": [
```
Comment (anacicconi) [inline .babelrc:3]: Very useful! I didn't know this plugin. Thanks @DanYellow

**Comment by anacicconi** [comment]
Thanks a lot for your help @DanYellow ! 

I agree. A webpack config for dev and prod is way more clear. 

About the less, I agree with you too. I added it because I'm currently working in a project that needs to import less stylesheets so I wanted to test it. However, I thought post-css-loader was useful only to add vendor prefixes to the css. Is it a substitute to less/sass too?

**Comment by anacicconi** [comment]
Oh, I almost forgot it. I thought about installing hot-reload. However, I don't see what it can add more than nodemon + babel watch + webpack watch. With the currently configuration, I can make changes to server js files, front js files and css and they are automatically loaded. Maybe I don't understand the idea behind hot-reload but I thought it was used to achieve this result, no?

**Comment by DanYellow** [comment]
So it's very simple to understand what hot reload brings to your project.
reload: reloads the entire browser
**react-**hot-reload: reloads only the component updated, it's very handful if you want to keep state of others components
webpack's hot reload: Reloads only the js modules and css updated (for css you need to disable ExtractTextPlugin)

**Pull request merged**